File: CH5(greedy)/spy.py
n = int(input())

for _ in range(n):
    s = input()

q = int(input())

agencies = set()
ans = 0

for _ in range(q):
    s = input()
    agencies.add(s)

    if len(agencies) == n:
        ans += 1
        agencies.clear()
        agencies.add(s)

print(ans)

File: CH5(greedy)/spy.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, q;
    string s;
    cin >> n;

    getline(cin, s); 
    for (int i = 0; i < n; i++) {
        getline(cin, s);
    }

    cin >> q;
    getline(cin, s); 

    unordered_set<string> agencies;
    int ans = 0;

    for (int i = 0; i < q; i++) {

        getline(cin, s);
        agencies.insert(s);

        if (agencies.size() == n) {
            ans++;

            agencies.clear();

            agencies.insert(s);
        }
    }
    cout << ans << '\n';
}